src/jobs/deploy_to_selfhosted.yml

```diff
@@ -29,6 +33,14 @@ steps:
             echo "https://app.circleci.com/pipelines/github/codacy/chart/$number/workflows/$id"
         }
 
+        function wait_for_workflow_started {
+            timeout 10s bash -c "while [[ \"$(echo $remoteWorkflow | jq -r 'isempty(.items[])')\" == "true" ]]; do sleep $4; echo \"$5\"; done" || false
+        }
+
+        function wait_for_workflow_terminated {
+            timeout 300s bash -c "while [[ \"$(echo $remoteWorkflow | jq -r '.items[0].status=="pending" or .items[0].status=="running"')\" == "true" ]]; do sleep 30s; echo \"Waiting for remote workflow to complete...\"; remoteWorkflow=$(get_workflow $workflowID); done" || false
```

Comment:
300s has been the upper limit duration for the remote workflow.

Reply:
with some buffer :)

Reply:
Now I see what is happening here, we are waiting for another CI to finish ...
Do we have another way for doing the same?

Reply:
Not that i know of. I mean we want the deploy to self hosted to show red on the workflow that invokes it as well right?
So it can not really be a fire and forget kind of thing. Otherwise we risk developers firing and forgetting a broken codacy installation on the self hosted dev environment :)